### PR TITLE
Add sample for other form input types

### PR DESCRIPTION
**Pull request by kul3r4**
Adds type=checkbox and type=date to the amp-form sample.
@sebastianbenz PTAL

### CODE REVIEW

File: src/20_Components/amp-form.html [hunk 1]
```diff
@@ -106,5 +106,26 @@
         </template>
     </div>
   </form>
+  <!-- #### Other form input samples -->
+  <!--
+    `amp-form` supports all HTML5 form types with the exception of file and password.
+     Use `type="date"` to allow the browser to display a native date picker. 
+  -->
+  <form method="post" action="<%host%>/components/amp-form/submit-form" target="_top">
```
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: I would provide an XHR action instead of changing the page.

**Comment by kul3r4** [comment]
@sebastianbenz PTAL

File: src/amp-form-success.html [hunk 2]
```diff
@@ -15,8 +15,7 @@
   <!--
   #### Success page for amp-form
 
-  Success! Thanks for trying amp-form demo. This page has been opened because you chose the amp-form with HTTP request.
-  Try to insert the word "error" as a name input in the form to see how amp-form handles errors. 
+  Success! Thanks for trying amp-form demo.
```
Comment (sebastianbenz) [inline src/amp-form-success.html]: s/trying/trying the/

File: src/20_Components/amp-form.html [hunk 3]
```diff
@@ -106,5 +111,46 @@
         </template>
     </div>
   </form>
+  <!-- #### Other form input samples -->
```
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: Can you also add samples for:

datetime-local
email
hidden
month
number
radio
range
tel
time
url
week
reset
search

I would put all inputs into a single form (also the checkbox one).

Also: What about image submit buttons? Are they allowed?

**Comment by kul3r4** [comment]
@sebastianbenz PTAL

File: src/20_Components/amp-form.html [hunk 4]
```diff
@@ -106,5 +118,88 @@
...
+  <!--
+    `amp-form` supports all HTML5 form types with the exception of file and password.
+     The following sample shows the supported types. 
+  -->
+  <form method="post" action="<%host%>/components/amp-form/submit-form" action-xhr="<%host%>/components/amp-form/submit-form-xhr" target="_top">
```
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: Can you increase the vertical margin between the form lines.

File: src/20_Components/amp-form.html [hunk 5]
```diff
@@ -106,5 +118,88 @@
...
+    <div class="vertical-display">
+      <div>
+          <label for="select-date">Select a date</label>
```
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: You're mixing two different kinds of labels, "Select a X" and "X".

**Comment by sebastianbenz** [comment]
Awesome!

I find the form a bit to noise as a result of the different labels. Why didn't you move the labels into the doc section? You could do something like this:

```
  <!--
    `amp-form` supports all HTML5 form types with the exception of file and password.
    The following sample shows the supported types. 
  -->
  <form method="post" 
    action="<%host%>/components/amp-form/submit-form" 
    action-xhr="<%host%>/components/amp-form/submit-form-xhr" 
    target="_top">
    <!-- Date -->
    <input name="select-date" type="date" value="2020-12-30">
    <!-- Month -->
    <input name="select-month" type="month" alue="2020-12-30">
    <!-- Week -->
    <input type="week" name="week_year">
    <!-- Date & Time -->
    <input name="select-datetime" type="datetime-local" value="2020-12-30">
    <!-- Time -->
    <input type="time" name="time_now">
    <!-- Checkbox -->
    <div>
      <input type="checkbox" name="animal1" value="Cats" checked> I like cats</input>
      <input type="checkbox" name="animal2" value="Dogs"> I like dogs </input>
    </div>
    <!-- Email -->
    <input type="email" name="email">
    <!-- Hidden fields -->
    <input type="hidden" name="city" value="London">
    <!-- Quantity -->
    <input type="number" name="quantity" min="1" max="5">
    <!-- Radio -->
    <div>
      <input type="radio" name="favourite animal" value="cat" checked> Cat<br>
      <input type="radio" name="favourite animal" value="dog"> Dog<br>
      <input type="radio" name="favourite animal" value="other"> Other
    </div>
    <!-- Range -->
    <input type="range" name="points" min="0" max="10">
    <!-- Telephone -->
    <input type="tel" name="my_tel">
    <!-- URL -->
    <input type="url" name="website">
    <!-- Search -->
    <input type="search" name="googlesearch">
    <!-- Reset -->
    <input type="reset" value="Reset" class="button button-primary">
    <!-- Submit -->
    <input type="submit" value="OK" class="button button-primary">
    <!-- Image -->
    <input type="image" src="/img/amp.jpg" alt="Submit Form" class="button button-primary button-image">

    <div submit-success>
      <template type="amp-mustache">
        Success! Thanks for trying the <code>amp-form</code> demo!
      </template>
    </div>
    <div submit-error>
      <template type="amp-mustache">
        Error!
      </template>
    </div>
  </form>
```
(I had to check if it actually works)

**Comment by kul3r4** [comment]
@sebastianbenz PTAL

File: src/20_Components/amp-form.html [hunk 6]
```diff
@@ -27,21 +27,32 @@
...
+  form.amp-form-submit-success [submit-success] {
+    color: green;
+  }
+  form.amp-form-submit-error [submit-error] {
+    color: red;
+  }
+  form.amp-form-submit-success.hide-inputs > input {
+    display: none
+  }
+  .vertical-display {
```
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: remove the no longer needed css

**Comment by sebastianbenz** [comment]
Awesome!!! One small nit + the range input doesn't seem to work in mobile mode.

**Comment by sebastianbenz** [comment]
LGTM

**Pull request merged**